homeassistant/components/device_tracker/__init__.py

```diff
@@ -460,6 +465,46 @@ def async_update(self):
             self._state = STATE_HOME
             self.last_update_home = True
 
+        # lookup vendor string for mac address using api.macvendors.com
+        if not self.vendor:
+            self.vendor = yield from self.get_vendor_for_mac()
```

Comment:
If the initial call (before writing to config) fails (not found, error, etc) it will return the value 'unknown' as vendor string. This will be written to config and it won't be retried again. There are test cases for this in the test suite.